nipoppy/extractors/networks/extract_network.py

```diff
@@ -0,0 +1,355 @@
+    """
+    Parse the inputs to create the DKT and Schaefer networks for the requested participant
+    """
+
+    # copy fmriprep config parsing - minor extensions for tractoflow needed
+
+    # extract paths
+    DATASET_ROOT = global_configs["DATASET_ROOT"]
+    TEMPLATEFLOW_DIR = global_configs["TEMPLATEFLOW_DIR"]
+    CONTAINER_STORE = global_configs["CONTAINER_STORE"]
+    FMRIPREP_CONTAINER = global_configs["PROC_PIPELINES"]["fmriprep"]["CONTAINER"]
+    TRACTOFLOW_CONTAINER = global_configs["PROC_PIPELINES"]["tractoflow"]["CONTAINER"]
+    FMRIPREP_VERSION = global_configs["PROC_PIPELINES"]["fmriprep"]["VERSION"]
+    TRACTOFLOW_VERSION = global_configs["PROC_PIPELINES"]["tractoflow"]["VERSION"]
+    FS_VERSION = global_configs["PROC_PIPELINES"]["freesurfer"]["VERSION"]
+    FMRIPREP_CONTAINER = FMRIPREP_CONTAINER.format(FMRIPREP_VERSION)
+    TRACTOFLOW_CONTAINER = TRACTOFLOW_CONTAINER.format(TRACTOFLOW_VERSION)
+
+    # pick the container - I'm not sure fMRIPrep has Dipy - TractoFlow container definitely has ANTs / Dipy
+    SINGULARITY_FMRIPREP = f"{CONTAINER_STORE}{FMRIPREP_CONTAINER}"
+    SINGULARITY_TRACTOFLOW = f"{CONTAINER_STORE}{TRACTOFLOW_CONTAINER}"
+
+    # set up logs
+    #log_dir = f"{DATASET_ROOT}/scratch/logs"
+
+    #if logger is None:
+    #    log_file = f"{log_dir}/network_extractor.log"
+    #    logger = my_logger.get_logger(log_file)
+
+    #logger.info("-"*75)
+    #logger.info(f"Using DATASET_ROOT: {DATASET_ROOT}")
+    #logger.info(f"Using participant_id: {participant_id}, session_id:{session_id}")
+    # logger.info(f"Optional args: --anat_only={anat_only}, --use_bids_filter={use_bids_filter}")
+    print(f"Using DATASET_ROOT: {DATASET_ROOT}")
+    print(f"Using PARTICIPANT_ID: {participant_id}, SESSION_ID: {session_id}")
+
+    if output_dir is None:
+        output_dir = f"{DATASET_ROOT}/derivatives/networks/v0.9.0"
+
+    if tractoflow_dir is None:
+        tractoflow_dir = f"{DATASET_ROOT}/derivatives/tractoflow/{TRACTOFLOW_VERSION}/output/ses-{session_id}"
+
+    if fmriprep_dir is None:
+        fmriprep_dir = f"{DATASET_ROOT}/derivatives/fmriprep/{FMRIPREP_VERSION}/output"
+
+    if freesurfer_dir is None:
+        freesurfer_dir = f"{DATASET_ROOT}/derivatives/freesurfer/{FS_VERSION}/output/ses-{session_id}"
+
+    # these paths build all the input / output files
+
+    # define paths to outputs
+    LABPATH = Path(output_dir)
+    SEGPATH = Path(output_dir, f"sub-{participant_id}", f"ses-{session_id}", 'anat')
+    NETPATH = Path(output_dir, f"sub-{participant_id}", f"ses-{session_id}", 'dwi')
+    # FUNPATH = Path(output_dir, f"sub-{participant_id}", f"ses-{session_id}", 'func')
+    # add functional as part of this? easy to load...
+
+    # make output paths if they don't exist
+    if ~os.path.exists(LABPATH):
+        os.makedirs(LABPATH)
+
+    if ~os.path.exists(SEGPATH):
+        os.makedirs(SEGPATH)
+
+    if ~os.path.exists(NETPATH):
+        os.makedirs(NETPATH)
```

Comment:
Use `pathlib` module to avoid `FileExistsError` resulting from preexisting dirs created by the initial run. 

Example:
`Path(some_dir).mkdir(parents=True, exist_ok=True)`